ctapipe/instrument/subarray.py

```diff
@@ -91,7 +91,9 @@ def __init__(
             coordinate system used for `tel_positions`.
         """
         self.name = name
-        self.positions = tel_positions or dict()
+        self.positions: Dict[int, Union[np.ndarray, EarthLocation]] = (
```

Comment:
We shouldn't use mixed types for internal storage. We should accept `EarthLocation` or `GroundFrame` coordinates for the inputs, but then convert in `__init__` to one common format.

I think for now, that should be the previous, so ground frame coordinas.